python/pyarrow/tests/test_parquet.py

```diff
@@ -1997,8 +1997,20 @@ def test_filters_inclusive_set(tempdir, use_legacy_dataset):
 
     dataset = pq.ParquetDataset(
         base_path, filesystem=fs,
-        filters=[('integer', 'in', {1}), ('string', 'in', {'a', 'b'}),
-                 ('boolean', 'in', {True})],
+        filters=[('string', 'in', 'ab')],
```

Comment:
I am wondering, should we raise an error here instead? Because now this is interpreted as `("string", "in", {"a", "b"})`, which I am not sure we should "silently" do (it's not very explicit for the user)

Reply:
('string', 'in', 'ab') will check if the partition in 'ab', so 'ab' will be accepted.
('string', 'in', {'a', 'b'})  will check if the partition in  {'a', 'b'}. And  'ab' will not be accepted.